### PR TITLE
Added keepDiacritics option.

### PR DESCRIPTION
**Pull request by philmtd**
This is related to issue #80 and allows to choose whether to keep diacritics/umlauts in links or not. The default behaviour is not changed and you need to actively decide not to replace the diacritics.

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -37,10 +39,10 @@ if (args._[0] !== '-') input = fs.createReadStream(args._[0]);
 
 input.pipe(utils.concat(function(input) {
   if (args.i) {
-    var newMarkdown = toc.insert(input.toString());
+    var newMarkdown = toc.insert(input.toString(), { keepDiacritics: args.keepDiacritics });
```
Comment (jonschlinkert) [inline cli.js]: here I think we should just pass `args`, so that all the options would be passed. Would you mind making that change?

Comment (philmtd) [inline cli.js]: Yep that makes sense, will do it in a sec.

**Comment by philmtd** [comment]
Now all cli `args` are passed to toc.

**Comment by philmtd** [comment]
@jonschlinkert Have you seen my update?

**Comment by jonschlinkert** [comment]
@philmtd sorry for the delay, thank you! I'll get this merged asap!

**Comment by philmtd** [comment]
@jonschlinkert thank you! :)

**Comment by philmtd** [comment]
Hey @jonschlinkert, do you think you could merge and release it this weekend? Would be awesome! :)

**Comment by jonschlinkert** [comment]
sure thing, thanks for the reminder. Sorry have had an extra busy week!

**Comment by philmtd** [comment]
Don't worry. I just "need" it next week so a handful of people can install it. And of course it would be easier for them via npm without having to pull my fork. So thanks in advance once again 😊

**Comment by philmtd** [comment]
Is anything going to happen here?

**Comment by jonschlinkert** [comment]
@philmtd sorry again, I'm trying to decide if `keepDiacritics` is the right name for this option. I think we need this, but since it's doing more than preserving diacritics (it's disabling escaping of query strings) maybe we should name it something like `escape` or `escapeQuerystring` instead? Which would be enabled by default. thoughts?

**Comment by philmtd** [comment]
Hi @jonschlinkert, yes I had similar thoughts about the name. My suggestion is to name it `escapeQuery`. I just made a commit with the changes and I think I like this better than `keepDiacritics`. What do you think?

**Comment by philmtd** [comment]
🔔 Hey @jonschlinkert. Are you going to merge this pull request anytime soon? I understand that you don't have much time but this is waiting for nearly two months now :(

**Comment by fabiohaertel** [comment]
Need this asap :/

**Comment by philmtd** [comment]
@fabiohaertel I already gave up on this PR a year ago.. Looks like @jonschlinkert abandoned this library.

**Comment by jonschlinkert** [comment]
> Looks like @jonschlinkert abandoned this library.

Interesting, I use this library 5-10 times a day when building readmes, and we've made commits since then. 

I apologize that your PR wasn't merged, but to be fair I have to focus my time on issues that impact the most users. I'm sure this would be useful to many users, but I can only go by the numbers. In the past year, before the comment from @fabiohaertel, you were the only person who commented on this issue, despite the library being downloaded 360k times since this issue was created, and this repository's issues are viewed by ~2k unique visitors a month. 

Not trying to be heavy-handed, I just want to point out why this hasn't got much attention. Only 1 (now 2) out of 25k people who viewed the issues on this repository mentioned a need for this feature.

**Comment by philmtd** [comment]
No hard feelings, I just moved on as I only needed this library for one project a year ago and since did not use or need it anyway.
I just think it wouldn't have had any impact on users who didn't need this extra option but would have helped those who needed it (like some GitLab users). 
I can absolutely understand your reasoning, though.

**Comment by tonprince** [comment]
I have the same issue, need umlaute in the toc.

**Comment by mwiede** [comment]
same issue here, what would be a workaround? the `filter` function?

**Pull request closed without merging**